### PR TITLE
[Yang model] Add Yang models for VNET table.

### PR DESCRIPTION
**Pull request by siqbal1986**
Created Yang Modle for VNET table.
https://github.com/sonic-net/sonic-buildimage/issues/14534

##### Work item tracking
- Microsoft ADO **(number only)**:
18215579
#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

File: src/sonic-yang-models/doc/Configuration.md [hunk 1]
```diff
@@ -2025,6 +2026,33 @@ channel name as object key, and tagging mode as attributes.
 }
 ```
 
+### VNET
+
+VLAN member table has Vnet name as the object key, and vxlan_tunnel name, scope, vni, peer list, advertised prefix, src mac, and overlay dest mac as attributes.
```
Comment (prsunny) [inline src/sonic-yang-models/doc/Configuration.md:2031]: VNET table (not VLAN).

File: src/sonic-yang-models/doc/Configuration.md [hunk 2]
```diff
@@ -2025,6 +2026,33 @@ channel name as object key, and tagging mode as attributes.
 }
 ```
 
+### VNET
+
+VLAN member table has Vnet name as the object key, and vxlan_tunnel name, scope, vni, peer list, advertised prefix, src mac, and overlay dest mac as attributes.
+The vxlan_tunnel name is the tunnel name from the VXLAN table. scope must "default", vni is the vxlan tunnel vni, peer_list is for Vnet
```
Comment (prsunny) [inline src/sonic-yang-models/doc/Configuration.md:2032]: Please mention scope can be optional

File: src/sonic-yang-models/yang-models/sonic-vnet.yang [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+                }
+
+                leaf guid {
+                    type string {
+                        length 1..255;
+                    }
+                }
+
+                leaf scope {
+                    description "can either be default or a valid VRF.";
```
Comment (prsunny) [inline src/sonic-yang-models/yang-models/sonic-vnet.yang:74]: Fix description. Can be only 'default'

File: src/sonic-yang-models/yang-models/sonic-vnet.yang [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+                    description "Flag to enable advertisement of route prefixes belonging to the Vnet.";
+                    type  boolean;
+                }
+
+                leaf overlay_dmac {
+                    description "Overlay Dest MAC address to be used by Vnet ping.";
+                    type yang:mac-address;
+                }
+
+                leaf src_mac {
```
Comment (prsunny) [inline src/sonic-yang-models/yang-models/sonic-vnet.yang:92]: Not sure if src_mac is part of the attribute, can you confirm?

Comment (siqbal1986) [reply src/sonic-yang-models/yang-models/sonic-vnet.yang:92]: We do handle it in the code.

File: src/sonic-yang-models/yang-models/sonic-vnet.yang (hunk 3 above)
Comment (siqbal1986) [reply src/sonic-yang-models/yang-models/sonic-vnet.yang]: done

File: src/sonic-yang-models/doc/Configuration.md (hunk 2 above)
Comment (siqbal1986) [reply src/sonic-yang-models/doc/Configuration.md]: done

File: src/sonic-yang-models/doc/Configuration.md (hunk 1 above)
Comment (siqbal1986) [reply src/sonic-yang-models/doc/Configuration.md]: done

File: src/sonic-yang-models/doc/Configuration.md [hunk 5]
```diff
@@ -2025,6 +2026,33 @@ channel name as object key, and tagging mode as attributes.
 }
 ```
 
+### VNET
+
+VNET member table has Vnet name as the object key, and vxlan_tunnel name, scope, vni, peer list, advertised prefix, src mac, and overlay dest mac as attributes.
```
Comment (prsunny) [inline src/sonic-yang-models/doc/Configuration.md:2031]: member? Change to VNET table

Comment (siqbal1986) [reply src/sonic-yang-models/doc/Configuration.md]: sorry, I missed that. its fixed now.

**Comment by siqbal1986** [comment]
@qiluo-msft can you please approve and merge this?

**Review by prsunny — DISMISSED** [review]

**Comment by siqbal1986** [comment]
@qiluo-msft can you please merge this

File: src/sonic-yang-models/yang-models/sonic-vnet.yang [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+        container VNET {
+
+            description "config db VNET table";
+
+            list VNET_LIST {
+
+                key "name";
+                
+                leaf name {
+                    type string;
```
Comment (ganglyu) [inline src/sonic-yang-models/yang-models/sonic-vnet.yang:46]: Can you add description?
Do we have any pattern for name?

File: src/sonic-yang-models/yang-models/sonic-vnet.yang [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+                }
+
+                leaf vxlan_tunnel {
```
Comment (ganglyu) [inline src/sonic-yang-models/yang-models/sonic-vnet.yang:49]: please add description.

File: src/sonic-yang-models/yang-models/sonic-vnet.yang [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+                    mandatory true;
+                    type leafref {
+                        path "/svxlan:sonic-vxlan/svxlan:VXLAN_TUNNEL/svxlan:VXLAN_TUNNEL_LIST/svxlan:name";
+                    }
+                }
+                
+                leaf vni {
```
Comment (ganglyu) [inline src/sonic-yang-models/yang-models/sonic-vnet.yang:56]: please add description.
We don't want to use leafref here?

File: src/sonic-yang-models/yang-models/sonic-vnet.yang [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+                    type stypes:vnid_type;
+                }
+
+                leaf peer_list {
+                    description "Set of peers";
+                    /* Values in leaf list are UNIQUE */
+                    type string;
+                }
+
+                leaf guid {
```
Comment (ganglyu) [inline src/sonic-yang-models/yang-models/sonic-vnet.yang:67]: please add description.

File: src/sonic-yang-models/yang-models/sonic-vnet.yang (hunk 6 above)
Comment (siqbal1986) [reply src/sonic-yang-models/yang-models/sonic-vnet.yang:47]: name can be any type of string. we dont have a specific requirement that it start with vnet*

Comment (siqbal1986) [reply src/sonic-yang-models/yang-models/sonic-vnet.yang:47]: description added.

File: src/sonic-yang-models/yang-models/sonic-vnet.yang (hunk 7 above)
Comment (siqbal1986) [reply src/sonic-yang-models/yang-models/sonic-vnet.yang:50]: description added.

File: src/sonic-yang-models/yang-models/sonic-vnet.yang (hunk 9 above)
Comment (siqbal1986) [reply src/sonic-yang-models/yang-models/sonic-vnet.yang:70]: done

**Review by prsunny — APPROVED** [review]

**Review by ganglyu — APPROVED** [review]
lgtm

**Comment by siqbal1986** [comment]
@qiluo-msft can you please merge this.

**Comment by yxieca** [comment]
@siqbal1986 can you help create cherry-pick PR for 202205 branch?

**Pull request merged**